File: 869-reordered-power-of-2/869-reordered-power-of-2.js
/**
 * @param {number} n
 * @return {boolean}
 */
var reorderedPowerOf2 = function(n) {
    let s = fun(n);
    for(let i=0;i<32;i++){
        if(s === fun(1<<i)) return 1;
    }
    return 0;
};

function fun(n){
    let str = n.toString();
    
    return str.split('').sort().join('');;
}

File: 869-reordered-power-of-2/869-reordered-power-of-2.cpp
class Solution {
public:
    bool reorderedPowerOf2(int n) {
        string s = sortit(n);
        for(int i=0;i<32;i++){
            if(s == sortit(1<<i)) return 1;
        }
        return 0;
    }
    string sortit(int n){
        string s = to_string(n);
        sort(s.begin(),s.end());
        return s;
    }
};